src/persist-client/src/write.rs

```diff
@@ -797,6 +805,15 @@ mod tests {
             .expect_compare_and_append_batch(&mut [&mut batch0, &mut batch1], 0, 4)
             .await;
 
+        let batch = write
+            .machine
+            .snapshot(&Antichain::from_elem(3))
+            .await
+            .expect("just wrote this")
+            .into_element();
+
+        assert!(batch.runs().count() >= 2);
```

Comment:
can we assert the specific runs? would also be useful to check cases of 1 batch having >1 run, >2 batches

Reply:
Humm... maybe time to try the datadriven thing then! Will see how it shakes out.